website/docs/developer/research/research-reconciliation.mdx

```diff
@@ -0,0 +1,683 @@
+              ...
+            ]
+          },
+          ...
+        ]
+      },
+      ...
+    ]
+  },
+  ...
+]
+```
+
+Some items we will need to think about include managing paging (start/stop) for
+large returns and limiting size of the message across the wire.
+
+
+### Potential ORCA Reconciliation Data Reporting Architecture
+
+The following provides a rough high level architecture for reporting the ORCA
+holdings to Cumulus. When providing data back to Cumulus, the solution should be
+able to handle the filters supplied and return back the needed data. In addition,
+the solution should be extensible and should allow for potential changes in the
+underlying data structure or storage engine as ORCA should be able to handle AWS
+PostgreSQL RDS, AWS Aurora PostgreSQL, and AWS Aurora PostgreSQL Serverless
+offerings.
+
+<MyImage
+    imageSource={useBaseUrl('img/Research-Reconciliation-Cumulus-Reconcile.png')}
+    imageAlt="System Context"
+    zoomInPic={useBaseUrl('img/zoom-in.svg')}
+    zoomOutPic={useBaseUrl('img/zoom-out.svg')}
+    resetPic={useBaseUrl('img/zoom-pan-reset.svg')}
+/>
+
+
+## Populating the metadata
+
+Populating the metadata objects with the proper metadata will need to come from
+various sources. The sections below will go through thoughts and ideas of where
+the metadata will originate from to populate the objects and potential ideas on
+what the underlying architecture may look like to perform the tasks.
+
+### Automated Cumulus Ingest Workflow
+
+The conventional way of populating the objects with metadata would be through
+the normal Cumulus ingest workflow that contains ORCA's `copy_to_glacier` lambda.
+The `copy_to_glacier` lambda would need to be modified to take or pull additional
+metadata needed for a granule's files like `providerId` and `collectionId` and
+various file information and send it to a queue. The body sent to the queue
+would have enough information for another ORCA lambda to write and/or update the
+metadata records for the various objects.
+
+Note that this new database write lambda or the `copy_to_glacier` lambda may need
+to make some additional queries to the S3 ORCA bucket to pull information like
+version and other values. Looking at the original `copy_to_glacier`, `s3.copy`
+is used to copy data to the ORCA archive. Based on initial research located
+[here](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.copy),
+the `s3.copy` command does not return any information. See the test performed
+below. Because of this, we will likely need to perform a listing on the object to
+get additional information like the version as seen in the sample seen below.
+Some things to consider when looking at the options
+for getting the additional data include speed (how fast can I get the data), cost
+(what is the cost impact of additional gets on the  s3 bucket), and complexity
+(where is the best place to put these additional gets, is it overly complex)
```

Comment:
(where is the best place to put these additional gets, is it overly complex).

Reply:
Reworked this section a bit to better clarify these as questions to ask during implementation.

Reply:
I would add question marks to the Qs and add a period after the last parenthesis...

<!--StartFragment-->Some things to consider when looking at the options for getting the additional data include speed (How fast can I get the data?), cost (What is the cost impact of additional gets on the  s3 bucket?), and complexity (Where is the best place to put these additional gets? Is it overly complex?).<!--EndFragment-->